sdk/src/transaction_context.rs

```diff
@@ -667,19 +842,111 @@ impl<'a> BorrowedAccount<'a> {
             )
             .unwrap_or_default()
     }
+
+    /// Returns true if the owner of this account is the current `InstructionContext`s last program (instruction wide)
+    pub fn is_owned_by_current_program(&self) -> bool {
+        self.instruction_context
+            .get_last_program_key(self.transaction_context)
+            .map(|key| key == self.get_owner())
+            .unwrap_or_default()
+    }
+
+    /// Returns an error if `get_data_mut` would fail
+    pub fn can_data_be_changed(&self) -> Result<(), InstructionError> {
+        if !self
```

Comment:
nit: I'm not a huge fan of this early exit behavior. Even after reviewing this PR for awhile I keep forgetting that this is a no-op if the feature isn't active. Can you pull out the feature gate check to the caller?

Reply:
You have to keep in min that at some point, the feature gate will be through and can be cleaned up. And the cleanup should be simple and its change set minimal, as it is itself a dangerous change. We already had issues with incorrectly removals of feature gate code. So, in that light I would prefer to keep these in the callee, as these have to many call sites which would all be affected.

Reply:
Ok, that's fine with me